istioctl/cmd/wait_test.go

```diff
@@ -74,41 +78,51 @@ func TestWaitCmd(t *testing.T) {
 		},
 		{
 			execClientConfig: cannedResponseMap,
-			args:             strings.Split("x wait --revision canary virtualservice foo.default", " "),
+			args:             strings.Split("x wait --timeout 2s --revision canary virtualservice foo.default", " "),
 			wantException:    false,
 		},
 	}
 
-	_ = setupK8Sfake()
-
 	for i, c := range cases {
 		t.Run(fmt.Sprintf("case %d %s", i, strings.Join(c.args, " ")), func(t *testing.T) {
+			_ = setupK8Sfake()
 			verifyExecTestOutput(t, c)
 		})
 	}
 }
 
 func setupK8Sfake() *fake.FakeDynamicClient {
-	objs := []runtime.Object{
-		newUnstructured("networking.istio.io/v1alpha3", "virtualservice", "default", "foo", "1"),
-		newUnstructured("networking.istio.io/v1alpha3", "virtualservice", "default", "bar", "3"),
-	}
-	client := fake.NewSimpleDynamicClient(runtime.NewScheme(), objs...)
+	client := fake.NewSimpleDynamicClient(runtime.NewScheme())
 	clientGetter = func(_, _ string) (dynamic.Interface, error) {
 		return client, nil
 	}
+	go func() {
+		// wait till watch created, then send create events.
+		// by default, k8s sends all existing objects at the beginning of a watch, but the test mock does not.  This
+		// function forces the test to behave like kubernetes does, but creates a race condition on watch creation.
+		time.Sleep(100 * time.Millisecond)
```

Comment:
This will result in slow and/or flaky tests, we shouldn't sleep

Reply:
We should either:
* Make the code resilient to resource not existing yet (helps real users)
* Make the watch code trigger that we can start creating resources. This can be done by adding some reactors on the fake client